api/swagger.yaml

```diff
@@ -4605,7 +4605,8 @@ definitions:
         example: false
       OOMKilled:
         description: |
-          Whether this container has been killed because it ran out of memory.
+          Whether a process within this container has been killed because it ran
+          out of memory since the container was last started.
```

Comment:
OOMKilled always meant "any process" was killed. I have only changed _when_ the flag is updated, and corrected the documentation.

Unfortunately, we can't reserve `OOMKilled` for the container's pid 1 because cgroups only keeps track of how many times the OOM killer has been invoked on processes in a group, not which processes in the group got killed. And even if we could, things could get really confusing with `docker run --init` and containers which use a fit-for-purpose init (e.g. `s6-svscan`) as pid 1. If we had the ability to know which processes got OOM-killed, I think it would be so much more useful to surface that information as discrete events associated with the container with timestamps, PIDs and `/proc/<pid>/cmdline`s.

Even without detailed information, I think surfacing OOMs as timestamped events would be far superior to a counter as it would allow the fatal OOM-kills to be differentiated from the non-fatal ones, even after the container exits. (Heuristic: how much time elapsed between the OOM-kill and the container stopping.) A close runner-up would be surfacing a counter along with the timestamp of the most recent OOM-killed event received by the daemon.

--------

If only [`memory.oom.group`](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#:~:text=memory.oom.group,of%20ancestor%20cgroups.) was enabled in containers. (AFAICT runC does not.) That would clear up any ambiguity quite nicely.